packages/esm-patient-attachments-app/src/attachments/use-allowed-extensions.ts

```diff
@@ -0,0 +1,31 @@
+import useSWRImmutable from 'swr/immutable';
+import { openmrsFetch } from '@openmrs/esm-framework';
+
+export interface GlobalProperty {
+  uuid: string;
+  property: string;
+  value: string;
+}
+
+/**
+ * Global property created by the openmrs-module-attachment
+ */
+const allowedExtensionsGlobalProperty: string = 'attachments.allowedFileExtensions';
+
+/**
+ * React hook that takes returns the allowed file extensions
+ * @returns String array containing the `allowedExtensions`, `isLoading` loading status, `error`
+ */
+export const useAllowedExtensions = () => {
+  const customRepresentation = 'custom:(uuid,property,value)';
+  const { data, error, isLoading } = useSWRImmutable<{ data: { results: Array<GlobalProperty> } }>(
+    `/ws/rest/v1/systemsetting?&v=${customRepresentation}&q=${allowedExtensionsGlobalProperty}`,
+    openmrsFetch,
+  );
+
+  return {
+    isLoading,
+    allowedExtensions: data?.data?.results?.length > 0 ? data?.data?.results[0].value?.split(',') : undefined,
```

Comment:
Deal with an empty string:

```suggestion
    allowedExtensions: data?.data?.results?.length > 0 ? data?.data?.results[0].value?.split(',') || undefined : undefined,
```